### PR TITLE
SwiftUI Cookbook

### PR DESCRIPTION
**Pull request by DmitriyHolovnia**
## Table of Contents:
- Keep your code clean and organized
- Configuring SwiftUI views
- Use ViewModifiers to reuse styling logic
- Use structs for data modeling
- Use environment objects for shared state
- Accesability

### CODE REVIEW

File: 11-swiftui.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    } 
+}
+```
+When creating a view with a lot of instances inside, it's important to keep performance in mind. Creating too many instances or having too complex of a view hierarchy can lead to slow performance, especially on older devices.
+
+To improve performance, you can follow these best practices:
+
+1. Use reusable views: If you have a lot of instances of the same view, consider creating a reusable view instead of creating a new instance for each item. For example, if you have a list of items, create a single item view and reuse it for each item.
+2.  Use lazy loading (`LazyVStack/LazyHStack` and `LazyHGrid/LazyVGrid`): If you have a lot of data to display, consider using lazy loading to only load and display the data that is currently visible on the screen. This can significantly improve performance, especially for large data sets.
+3.  Avoid complex view hierarchies: Try to keep your view hierarchy as simple as possible. If you have a complex view hierarchy, consider breaking it down into smaller views or using a different layout. Avoid computed `View` property.
```
Comment (AlexanderKulyk) [inline 11-swiftui.md:39]: Avoiding computed `View` properties contradicts the 
> Also, separate your code into logical blocks using extensions or **subviews**.

guideline. Let's add examples for the logical blocks separation guideline to make it more clear.

File: 11-swiftui.md [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+        }
+    }
+}
+```
+---
+
+### Use environment objects for shared state
+Environment objects are a way to share state between views in your app. Use environment objects to manage shared state, such as user preferences or app settings.
+```swift
+class UserSettings: ObservableObject {
```
Comment (AlexanderKulyk) [inline 11-swiftui.md:189]: nit: Lets make it `final` to show our commitment to keeping things `final` where possible 😅

File: 11-swiftui.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+        List(people) {
+            person in Text(person.name) 
+        }
+    }
+}
+```
+---
+
+### Use environment objects for shared state
+Environment objects are a way to share state between views in your app. Use environment objects to manage shared state, such as user preferences or app settings.
```
Comment (AlexanderKulyk) [inline 11-swiftui.md:187]: Let's add a limitation: to use `@EnvironmentObject` for app-wide purposes only and prefer `@ObservedObject` and `@StateObject` for VMs and view-specific logic items.

File: 11-swiftui.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+    }
+}
+
+struct HomeContentView: View {
+    var text: String
+
+    var body: some View {
+        VStack {
+            Text(text)
+        }
```
Comment (AlexanderKulyk) [inline 11-swiftui.md:67]: This example gives me an idea of "healthy nesting". If and only if nesting > 4 we need to refactor any further nesting into a dedicated view. This way we will not create too many small unique views.

**Review by MykhailoPalchuk — COMMENTED** [review]
Nice doc! Could you please add a list of books/resources you would recommend someone who's learning SwiftUI? I think that would be a nice addition to this doc

**Review by ANDaleksei — APPROVED** [review]
Nice!

**Review by OleksandrKulyk — APPROVED** [review]
👍 
I updated tabulation to match other markdowns in our repo. 
Looks great both content-wise and in Github file preview -> https://github.com/uptechteam/ios-cookbook/blob/dmitriy/swiftui/11-swiftui.md

**Pull request merged**